Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -91,59 +93,141 @@ public void onResult(@Nullable Configuration configuration, @Nullable Exception
                     return;
                 }
 
-                sharedPrefsWriter.persistVenmoVaultOption(activity, request.getShouldVault() && braintreeClient.getAuthorization() instanceof ClientToken);
+                if (request.getPaymentMethodUsage() == VenmoPaymentMethodUsage.UNSPECIFIED) {
+                    braintreeClient.sendAnalyticsEvent("pay-with-venmo.app-switch.failed");
+                    callback.onResult(new BraintreeException("Payment method usage must be set."));
+                }
 
                 String venmoProfileId = request.getProfileId();
                 if (TextUtils.isEmpty(venmoProfileId)) {
                     venmoProfileId = configuration.getVenmoMerchantId();
                 }
-                activity.startActivityForResult(getLaunchIntent(configuration, venmoProfileId),
-                        BraintreeRequestCodes.VENMO);
-                braintreeClient.sendAnalyticsEvent("pay-with-venmo.app-switch.started");
+
+                JSONObject params = new JSONObject();
+                try {
+                    params.put("query", "mutation CreateVenmoPaymentContext($input: CreateVenmoPaymentContextInput!) { createVenmoPaymentContext(input: $input) { venmoPaymentContext { id } } }");
+                    JSONObject input = new JSONObject();
+                    input.put("paymentMethodUsage", request.getPaymentMethodUsageAsString());
+                    input.put("merchantProfileId", venmoProfileId);
+                    input.put("customerClient", "MOBILE_APP");
+                    input.put("intent", "CONTINUE");
+                    JSONObject variables = new JSONObject();
+                    variables.put("input", input);
+                    params.put("variables", variables);
+                } catch (JSONException e) {
+                    callback.onResult(new BraintreeException("unexpected error"));
+                }
+
+                final String finalVenmoProfileId = venmoProfileId;
+                braintreeClient.sendGraphQLPOST(params.toString(), new HttpResponseCallback() {
+                    @Override
+                    public void success(String responseBody) {
+                        String paymentContextId = parsePaymentContextId(responseBody);
+                        if (paymentContextId == null || TextUtils.isEmpty(paymentContextId)) {
```

Comment:
```suggestion
                        if (TextUtils.isEmpty(paymentContextId)) {
```
I believe `TextUtils.isEmpty` will handle the null check.

Reply:
Yes good call